### PR TITLE
Feat: Withdrawal Status modal

### PR DESCRIPTION
**Pull request by malaniz**
Feat: Withdrawal Status modal

img-BYq0Cg_000

### CODE REVIEW

**Review by cazala — CHANGES_REQUESTED** [review]

File: src/components/Modals/WithdrawalStatusModal/WithdrawalStatusModal.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        closeIcon={<Close onClick={onClose} />}
+        className="WithdrawalStatusModal"
+      >
+        <Modal.Header>{t('withdrawal_status_modal.title')}</Modal.Header>
+        <Modal.Content>
+          <div className="amount_placeholder">
+            {t('withdrawal_status_modal.amount_placeholder')}
+          </div>
+          <div className="amount">
+            {amount} {t('global.usd_symbol')}
```
[the remark this answers is not in the document]
Comment (cazala) [reply src/components/Modals/WithdrawalStatusModal/WithdrawalStatusModal.tsx:40]: We could show the USD value underneath the mana amount, like we do on the ConvertManaModal, but this is mixing the MANA amount with the USD symbol

File: src/components/Modals/WithdrawalStatusModal/WithdrawalStatusModal.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+              label={t('withdrawal_status_modal.status_completed')}
+            />
+          </div>
+          <Button
+            primary
+            disabled={isPending || isLoading}
+            loading={isLoading}
+            onClick={handleFinishWithdrawal}
+          >
+            {t('global.done')}
```
Comment (cazala) [inline src/components/Modals/WithdrawalStatusModal/WithdrawalStatusModal.tsx:71]: This button's copy should be `Complete Withdrawal` while the status is `pending` or `checkpoint`, and `Done` once the status is `complete`. 

The handler should trigger a `onFinishWithdrawal(withdrawal)` only when the status is `checkpoint`, and a `onClose()` if it's on status `complete`

**Review by cazala — CHANGES_REQUESTED** [review]

File: src/components/Modals/WithdrawalStatusModal/WithdrawalStatusModal.tsx (hunk 2 above)
Comment (malaniz) [reply src/components/Modals/WithdrawalStatusModal/WithdrawalStatusModal.tsx]: totally agree. Fixing it right now.

File: src/components/Modals/WithdrawalStatusModal/WithdrawalStatusModal.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    if (!withdrawal) {
+      return
+    }
+    const { status, amount } = withdrawal
+    const isWithdrawalPending = status === WithdrawalStatus.PENDING
+    const isWithdrawalCheckpoint = status === WithdrawalStatus.CHECKPOINT
+    const isWithdrawalCompleted = status === WithdrawalStatus.COMPLETE
+
+    const handleFinishWithdrawal = () => onFinishWithdrawal(withdrawal)
+    const handleCloseModal = () => closeModal('WithdrawalStatusModal')
```
Comment (cazala) [inline src/components/Modals/WithdrawalStatusModal/WithdrawalStatusModal.tsx:39]: this should be `const handleCloseModal = () => onClose()`

**Review by cazala — CHANGES_REQUESTED** [review]

File: src/components/Modals/WithdrawalStatusModal/WithdrawalStatusModal.tsx (hunk 3 above)
Comment (malaniz) [reply src/components/Modals/WithdrawalStatusModal/WithdrawalStatusModal.tsx]: Thanks @cazala ! fixed!

**Review by cazala — APPROVED** [review]

**Pull request merged**